lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -304,16 +304,45 @@ class DetailsRenderer {
     }
 
     return tableLike.headings.map(heading => {
+      let multi;
+      if (heading.multi) {
+        multi = {
+          key: heading.multi.key,
+          valueType: heading.multi.itemType,
+          displayUnit: heading.displayUnit,
+          granularity: heading.granularity,
+        };
+      }
+
       return {
         key: heading.key,
         label: heading.text,
         valueType: heading.itemType,
+        multi,
         displayUnit: heading.displayUnit,
         granularity: heading.granularity,
       };
     });
   }
 
+  /**
+   * @param {LH.Audit.Details.TableItem} row
+   * @param {LH.Audit.Details.OpportunityColumnHeading} heading
+   * @return {Element?}
+   */
+  _renderMultiValue(row, heading) {
+    const values = row[heading.key];
+    if (!Array.isArray(values)) return null;
+    const valueElement = this._dom.createElement('div', 'lh-multi-values');
+    for (const childValue of values) {
+      const childValueElement = this._renderTableValue(childValue, heading);
```

Comment:
does this actually work with arbitrary value types or is report css only setup to accept text/url/code/numbers?

Reply:
I'd expect it to work for any `Value` types, although I've only used what you listed so far. For `code`, the default `pre` browser styles had to be tweaked (way too much whitespace). maybe others would need the same.